src/Microsoft.OpenApi.OData.Reader/Operation/MediaEntityDeleteOperationHandler.cs

```diff
@@ -0,0 +1,127 @@
+
+        protected override void Initialize(ODataContext context, ODataPath path)
+        {
+            base.Initialize(context, path);
+
+            if (Property != null)
+            {
+                _deleteRestrictions = Context.Model.GetRecord<DeleteRestrictionsType>(TargetPath, CapabilitiesConstants.DeleteRestrictions);
+                if (Property is IEdmNavigationProperty)
+                {
+                    var navigationDeleteRestrictions = Context.Model.GetRecord<NavigationRestrictionsType>(Property, CapabilitiesConstants.NavigationRestrictions)?
+                            .RestrictedProperties?.FirstOrDefault()?.DeleteRestrictions;
+                    if (_deleteRestrictions != null && navigationDeleteRestrictions != null)
+                    {
+                        _deleteRestrictions.MergePropertiesIfNull(navigationDeleteRestrictions);
+                    }
+                    _deleteRestrictions ??= navigationDeleteRestrictions;
+                }
+                else
+                {
+                    var propertyDeleteRestrictions = Context.Model.GetRecord<DeleteRestrictionsType>(Property, CapabilitiesConstants.DeleteRestrictions);
+                    if (_deleteRestrictions != null && propertyDeleteRestrictions != null)
+                    {
+                        _deleteRestrictions.MergePropertiesIfNull(propertyDeleteRestrictions);
+                    }
+                    _deleteRestrictions ??= propertyDeleteRestrictions;
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void SetBasicInfo(OpenApiOperation operation)
+        {
+            // Summary
+            string placeholderValue = LastSegmentIsStreamPropertySegment ? Path.LastSegment.Identifier : "media content";
+            operation.Summary = _deleteRestrictions?.Description;
+            operation.Summary ??= IsNavigationPropertyPath
+                ? $"Delete {placeholderValue} for the navigation property {NavigationProperty.Name} in {NavigationSourceSegment.NavigationSource.Name}"
+                : $"Delete {placeholderValue} for {NavigationSourceSegment.EntityType.Name} in {NavigationSourceSegment.Identifier}";
+
+            // Description
+            operation.Description = _deleteRestrictions?.LongDescription ?? Context.Model.GetDescriptionAnnotation(Property);
+
+            // OperationId
+            if (Context.Settings.EnableOperationId)
+            {
+                string identifier = LastSegmentIsStreamPropertySegment ? Path.LastSegment.Identifier : "Content";
+                operation.OperationId = GetOperationId("Delete", identifier);
+            }
+
+            base.SetBasicInfo(operation);
+        }
+
+        /// <inheritdoc/>
+        protected override void SetParameters(OpenApiOperation operation)
+        {
+            base.SetParameters(operation);
+
+            operation.Parameters.Add(new OpenApiParameter
```

Comment:
out of curiosity, is there a reason the `"If-Match"` header is added by default?